src/ol/controls/RoutingControl.js

```diff
@@ -0,0 +1,419 @@
+import { Vector } from 'ol/layer';
+import { Circle, Stroke, Style, Fill } from 'ol/style';
+import { Vector as VectorSource } from 'ol/source';
+import { Feature } from 'ol';
+import { LineString, Point } from 'ol/geom';
+import { Modify } from 'ol/interaction';
+import { unByKey } from 'ol/Observable';
+import { click } from 'ol/events/condition';
+import { fromLonLat, toLonLat } from 'ol/proj';
+import lineSlice from '@turf/line-slice';
+import { getDistance } from 'ol/sphere';
+import {
+  lineString as createTurfLine,
+  point as createTurfPoint,
+} from '@turf/helpers';
+import RoutingAPI from '../../api/routing/RoutingAPI';
+import Control from '../../common/controls/Control';
+import Layer from '../layers/Layer';
+
+const defaultPointStyle = new Style({
+  image: new Circle({
+    radius: 6,
+    fill: new Fill({
+      color: [255, 0, 0, 1],
+    }),
+  }),
+});
+
+/**
+ * Display layer's copyrights.
+ *
+ * @example
+ * import { Map, RoutingControl } from 'mobility-toolbox-js/ol';
+ *
+ * const map = new Map({
+ *   target: 'map',
+ *   controls: [
+ *     new RoutingControl()
+ *   ]
+ * });
+ *
+ *
+ * @see <a href="/example/ol-routing">Openlayers routing example</a>
+ *
+ * @extends {Control}
+ * @implements {RoutingInterface}
+ */
+class RoutingControl extends Control {
+  constructor(options) {
+    const opts = {
+      active: true,
+      ...options,
+    };
+
+    super(opts);
+
+    this.abortController = new AbortController();
+
+    this.apiKey = opts.apiKey;
+
+    this.api = new RoutingAPI({
+      apiKey: this.apiKey,
+    });
+
+    this.mot = opts.mot || 'bus';
+
+    this.routingLayer =
```

Comment:
How about extracting this into a `RouteLayer` class?

Reply:
Done